src/handlers/gestures/gesture.ts

```diff
@@ -0,0 +1,289 @@
+  simultaneousWith?: (
+    | number
+    | BaseGesture<unknown>
+    | React.RefObject<BaseGesture<unknown>>
+  )[];
+}
+
+type HandlerCallbacks<EventPayloadT> = {
+  handlerTag: number;
+  onBegan?: (
+    event: UnwrappedGestureHandlerStateChangeEvent<EventPayloadT>
+  ) => void;
+  onStart?: (
+    event: UnwrappedGestureHandlerStateChangeEvent<EventPayloadT>
+  ) => void;
+  onEnd?: (
+    event: UnwrappedGestureHandlerStateChangeEvent<EventPayloadT>,
+    success: boolean
+  ) => void;
+  onUpdate?: (event: UnwrappedGestureHandlerEvent<EventPayloadT>) => void;
+};
+
+export abstract class Gesture {
+  /**
+   * Return array of gestures, providing the same interface for creating and updating
+   * handlers, no matter which object was used to create gesture instance.
+   */
+  abstract configure(): BaseGesture<any>[];
+
+  /**
+   * Assign handlerTag to the gesture instance and set ref.current (if a ref is set)
+   */
+  abstract initialize(): void;
+
+  /**
+   * Make sure that values of properties defining relations are arrays. Do any necessary
+   * preprocessing required to configure relations between handlers. Called just before
+   * updating the handler on the native side.
+   */
+  abstract prepare(): void;
+}
+
+export abstract class BaseGesture<EventPayloadT> extends Gesture {
+  public handlerTag = -1;
+  public handlerName = '';
+  public config: BaseGestureConfig = {};
+  public handlers: HandlerCallbacks<EventPayloadT> = {
+    handlerTag: -1,
+  };
+
+  private addDependency(
+    key: 'simultaneousWith' | 'requireToFail',
+    gesture: BaseGesture<unknown> | React.RefObject<BaseGesture<unknown>>
+  ) {
+    this.config[key] = this.config[key]
+      ? [].concat(this.config[key] as never, gesture as never)
+      : [gesture];
+  }
+
+  setRef(ref: React.MutableRefObject<BaseGesture<unknown>>) {
+    this.config.ref = ref;
+    return this;
+  }
+
+  setOnBegan(
+    callback: (
+      event: UnwrappedGestureHandlerStateChangeEvent<EventPayloadT>
+    ) => void
+  ) {
+    this.handlers.onBegan = callback;
+    return this;
+  }
+
+  setOnStart(
+    callback: (
+      event: UnwrappedGestureHandlerStateChangeEvent<EventPayloadT>
+    ) => void
+  ) {
+    this.handlers.onStart = callback;
+    return this;
+  }
+
+  setOnEnd(
+    callback: (
+      event: UnwrappedGestureHandlerStateChangeEvent<EventPayloadT>,
+      success: boolean
+    ) => void
+  ) {
+    this.handlers.onEnd = callback;
+    return this;
+  }
+
+  setEnabled(enabled: boolean) {
+    this.config.enabled = enabled;
+    return this;
+  }
+
+  setMinPointers(minPointers: number) {
+    this.config.minPointers = minPointers;
+    return this;
+  }
+
+  setShouldCancelWhenOutside(value: boolean) {
+    this.config.shouldCancelWhenOutside = value;
+    return this;
+  }
+
+  setHitSlop(hitSlop: HitSlop) {
+    this.config.hitSlop = hitSlop;
+    return this;
+  }
+
+  addSimultaneousGesture(
+    gesture: BaseGesture<any> | React.RefObject<BaseGesture<any>>
```

Comment:
`any`

Reply:
Fixed in 6da30cb388f462568017bb46206988886afd142d